model_compression_toolkit/gptq/pytorch/gptq_training.py

```diff
@@ -276,6 +289,40 @@ def micro_training_loop(self,
                         self.loss_list.append(loss_value.item())
                         Logger.debug(f'last loss value: {self.loss_list[-1]}')
 
+    def _get_loss_weights(self, input_tensors: List[torch.Tensor]) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Fetches weights for distillation and round regularization parts of loss.
+
+        Args:
+            input_tensors: list containing a batch of inputs.
+
+        Returns:
+            A tuple of two tensors:
+            - weights for distillation loss
+            - weights for rounding regularization loss
+
+        """
+        if self.use_sample_layer_attention is False:
+            return self.hessian_score_per_layer, torch.ones_like(self.hessian_score_per_layer)
+
+        if len(input_tensors) > 1:
+            raise NotImplementedError('Sample-Layer attention is not currently supported for networks with multiple inputs')
+
+        scores = []
+        batch = input_tensors[0]
+        img_hashes = [self.hessian_service.calc_image_hash(img) for img in batch]
+        for img_hash in img_hashes:
+            if img_hash not in self.hessian_score_per_image_per_layer:
+                score_per_image_layer_per = self._compute_sample_layer_attention_scores(input_tensors)
```

Comment:
I guess you meant "score_per_image_per_layer"?
In general, I think this function needs to be rewritten cause it's hard to track what's going on here.

Reply:
Everything should be rewritten. Added more comments, hope it's clearer now.